### DISCUSSION TITLE
How to ignore line separator character like <0x2028> when using smart_open and unzipping the file

### DISCUSSION THREAD

**Discussion by Kajalagg98** [RaRe-Technologies/smart_open#761] [Q&A] [answered]
I have file in which there exits event like this, Line 1
`"_raw":"Relative Target Name:\tosadirkhanov\\Desktop\\Iltt Sxm - Now on island_ the Official Johnnie Walker Black Label St. Maarten Edition To keep you walking strong in 2019, we're giving away 11 of these limited edition bottles!! TO WIN_ **<0x2028>** 1. LIKE _files\\folder.jpg\r\r\n\r\r\nAccess Request Information:\r\r\n\tAccess Mask:\t\t0x80\r\r\n\tAccesses:\t\tReadAttributes\r\r\n\t\t\t\t\r\r\nAccess Check Results:\r\r\n\tReadAttributes:\tGranted by\tD:(A;;FA;;;WD)"`

On unzipping and reading this file using smart_open, the above event is separated into two events(Line 1 and Line2) like this:
`"_raw":"Relative Target Name:\tosadirkhanov\\Desktop\\Iltt Sxm - Now on island_ the Official Johnnie Walker Black Label St. Maarten Edition To keep you walking strong in 2019, we're giving away 11 of these limited edition bottles!! TO WIN_ **<0x2028>**`
 
 `1. LIKE _files\\folder.jpg\r\r\n\r\r\nAccess Request Information:\r\r\n\tAccess Mask:\t\t0x80\r\r\n\tAccesses:\t\tReadAttributes\r\r\n\t\t\t\t\r\r\nAccess Check Results:\r\r\n\tReadAttributes:\tGranted by\tD:(A;;FA;;;WD)"`

What I am looking for is to ignore this line separator so that single event won't get separated into two events.
I am using smart_open version 1.9.0. The constraint here python version 3.7 which I am currently using and is the demand of my project.

Any help would be appreciated?
Thanks.

**Comment by mpenkov** [comment, marked as answer]
Read the file as binary and perform the line splitting yourself (on the newline byte).

**Reply by Kajalagg98** [reply]
But I want to avoid the line splitting which is currently happening because of this line separator character in the text?

**Reply by mpenkov** [reply]
Yes, and that's why you have to perform the line splitting yourself, using your own criteria for what a line break should be (instead of letting the standard library do it for you).

**Reply by Kajalagg98** [reply]
Thanks, Michael.
